### PR TITLE
NAS-130121 / 24.10 / Render GPU feature

### PR DESCRIPTION
**Pull request by sonicaj**
This PR adds changes so GPU refs are properly rendered.

### CODE REVIEW

**Review by yocalebo — APPROVED** [review]

File: catalog_reader/questions.py [hunk 1]
```diff
@@ -35,23 +35,74 @@ def normalize_question(question: dict, version_data: dict, context: dict) -> Non
...
+                g['vendor'] == 'NVIDIA' and g['vendor_specific_config'].get('uuid') for g in gpu_choices
+            )
             data['attrs'] = [
                 {
-                    'variable': gpu,
-                    'label': f'GPU Resource ({gpu})',
-                    'description': 'Please enter the number of GPUs to allocate',
+                    'variable': 'use_all_gpus',
+                    'label': 'Passthrough available GPUs',
+                    'description': 'Please select this option to passthrough all available GPUs to the app',
```
Comment (stavros-k) [inline catalog_reader/questions.py:56]: I think we should rephrase this a bit.

Nvidia gpus can be passed independently of intel/amd.
So even if user selects `use_all_gpus` user can still pick a specific nvidia gpu to pass.

So proposing to rephrase
"Passthrough available (non-NVIDIA) GPUs"
"Please select this option to passthrough all available (non-NVIDIA) GPUs to the app"

Comment (stavros-k) [reply catalog_reader/questions.py:56]: Unless we want to pass actually all gpus here (including nvidia).
Then we should add a `show_if: [["use_all_gpus", "=", false]]` on the `nvidia_gpu_selection`

Comment (sonicaj) [reply catalog_reader/questions.py:56]: @stavros-k is it possible on a system having AMD/INTEL and NVIDIA GPUs, and you are only passing through AMD/INTEL gpus ?

Comment (stavros-k) [reply catalog_reader/questions.py:56]: Yes, amd/intel are passed via `/dev/dri` while nvidia via complete other definition

**Review by stavros-k — APPROVED** [review]

File: catalog_reader/questions.py (hunk 1 above)
Comment (sonicaj) [reply catalog_reader/questions.py]: @stavros-k so the intended behavior now is that if the boolean for use_all_gpus is set, all non-nvidia gpu's will be passed through

Comment (stavros-k) [reply catalog_reader/questions.py]: Correct! Thanks

**Pull request merged**